notebooks/0110_cmip-conc-inversions.py

```diff
@@ -0,0 +1,337 @@
+    "HFC43-10": Q(17.0, "yr"),
+    "NF3": Q(569.0, "yr"),
+    "SF6": Q((850 + 1280) / 2.0, "yr"),
+    "SO2F2": Q(36.0, "yr"),
+    "cC4F8": Q(3200.0, "yr"),
+}
+molecular_masses = {
+    # Gas: molecular mass
+    "CF4": Q(12.01 + 4 * 19.0, "gCF4 / mole"),
+    "C2F6": Q(2 * 12.01 + 6 * 19.0, "gC2F6 / mole"),
+    "C3F8": Q(3 * 12.01 + 8 * 19.0, "gC3F8 / mole"),
+    "C4F10": Q(4 * 12.01 + 10 * 19.0, "gC4F10 / mole"),
+    "C5F12": Q(5 * 12.01 + 12 * 19.0, "gC5F12 / mole"),
+    "C6F14": Q(6 * 12.01 + 14 * 19.0, "gC6F14 / mole"),
+    "C7F16": Q(7 * 12.01 + 16 * 19.0, "gC7F16 / mole"),
+    "C8F18": Q(8 * 12.01 + 18 * 19.0, "gC8F18 / mole"),
+    "CH2Cl2": Q(12.01 + 2 * 1.0 + 2 * 35.45, "gCH2Cl2 / mole"),
+    "CH3Br": Q(12.01 + 3 * 1.0 + 79.90, "gCH3Br / mole"),
+    "CH3Cl": Q(12.01 + 3 * 1.0 + 35.45, "gCH3Cl / mole"),
+    "CHCl3": Q(12.01 + 1.0 + 3 * 35.45, "gCHCl3 / mole"),
+    "HFC134a": Q(12.01 + 2 * 1.0 + 19.0 + 12.01 + 3 * 19.0, "gHFC134a / mole"),  # CH2FCF3
+    "HFC152a": Q(12.01 + 3 * 1.0 + 12.01 + 1.0 + 2 * 19.0, "gHFC152a / mole"),  # CH3CHF2
+    "HFC236fa": Q(12.01 + 3 * 19.0 + 12.01 + 2 * 1.0 + 12.01 + 3 * 19.0, "gHFC236fa / mole"),  # CF3CH2CF3
+    "HFC245fa": Q(
+        12.01 + 2 * 1.0 + 19.0 + 12.01 + 2 * 19.0 + 12.01 + 1.0 + 2 * 19.0, "gHFC245fa / mole"
+    ),  # CH2FCF2CHF2
+    "HFC365mfc": Q(
+        12.01 + 3 * 1.0 + 12.01 + 2 * 19.0 + 12.01 + 2 * 1.0 + 12.01 + 3 * 19.0, "gHFC365mfc / mole"
+    ),  # CH3CF2CH2CF3
+    "HFC43-10": Q(
+        12.01 + 3 * 19.0 + 2 * (12.01 + 1.0 + 19.0) + 12.01 + 2 * 19.0 + 12.01 + 3 * 19.0, "gHFC4310 / mole"
+    ),  # CF3CHFCHFCF2CF3
+    "NF3": Q(14.01 + 3 * 19.0, "gNF3 / mole"),
+    "SF6": Q(32.07 + 6 * 19.0, "gSF6 / mole"),
+    "SO2F2": Q(32.07 + 2 * 16.0 + 2 * 19.0, "gSO2F2 / mole"),
+    "cC4F8": Q(4 * 12.01 + 8 * 19.0, "gcC4F8 / mole"),
+}
+
+# %%
+# CDIAC https://web.archive.org/web/20170118004650/http://cdiac.ornl.gov/pns/convert.html
+ATMOSPHERE_MASS = Q(5.137 * 10**18, "kg")
+# https://www.engineeringtoolbox.com/molecular-mass-air-d_679.html
+MOLAR_MASS_DRY_AIR = Q(28.9, "g / mol")
+atm_moles = (ATMOSPHERE_MASS / MOLAR_MASS_DRY_AIR).to("mole")
+# Lines up with CDIAC: https://web.archive.org/web/20170118004650/http://cdiac.ornl.gov/pns/convert.html
+fraction_factor = Q(1e-6, "1 / ppm")
+mass_one_ppm_co2 = atm_moles * fraction_factor * Q(12.01, "gC / mole")
+cdiac_expected = 2.13
+if np.round(mass_one_ppm_co2.to("GtC / ppm").m, 2) != cdiac_expected:
+    raise AssertionError
+
+# %%
+assumed_unit = "ppt"
+background_emissions_d = {}
+inverse_info_l = []
+out_ts_vars = [*FOLLOW_LEADERS.keys(), "Emissions|C6F14"]
+out_pi_vars = [
+    "Emissions|CF4",
+    "Emissions|C2F6",
+    "Emissions|HFC|HFC152a",
+    "Emissions|HFC|HFC236fa",
+    "Emissions|HFC|HFC365mfc",
+    "Emissions|HFC|HFC134a",
+    "Emissions|HFC|HFC245fa",
+    "Emissions|HFC|HFC43-10",
+    "Emissions|SF6",
+]
+pi_year = 1750
+for emissions_var in tqdman.tqdm(sorted([*out_ts_vars, *out_pi_vars])):
```

Comment:
here, use `for emissions_var in tqdman.tqdm(sorted(list(set([*out_ts_vars, *out_pi_vars])))):` the conversion to set and back to list takes care of a few duplicates in the output file

Reply:
Nice, thanks (I assume you're working on this in another branch so I'll just let you make the change there rather than making it here too?)

Reply:
Yeah - also I'm going to invert everything that's sensible to invert, and decide later if we want to use it or not

Reply:
nice one

Reply:
Not sure where you're up to @chrisroadmap and Jarmo wants to move so I've pulled this in here. We can figure out merge conflicts later.